Containerfile

```
@@ -0,0 +1,41 @@
+ARG GO_VERSION=1.23
+FROM golang:${GO_VERSION} AS base
+
+# Cache dependencies:
+# The go mod download command uses a cache mount,
+# ensuring Go modules are cached separately from the build context and not included in image layers.
+# This cache is used in the build stage and reused across builds, unless go.mod or go.sum changes.
+WORKDIR /build
+COPY go.mod go.sum ./
+RUN --mount=type=cache,target=/go/pkg/mod \
+    go mod download
+
+FROM base AS build
+
+# Mount source code and build:
+# The --mount=target=. option mounts the source code without adding an extra image layer, unlike `COPY . .`.
+# The go build command uses the dependency cache and a dedicated mount to cache build artifacts for future builds.
+RUN --mount=target=. \
+    --mount=type=cache,target=/go/pkg/mod \
+    --mount=type=cache,target=/root/.cache/go-build \
+    CGO_ENABLED=0 GOOS=linux go build -trimpath -ldflags '-s -w' -o /icingadb ./cmd/icingadb/main.go
```

By specifying the `main.go` file - which is unnecessary in this case -, the VCS information are missing.

```
docker run -it --rm icingadb /icingadb --version
Icinga DB version: 1.2.1

Build information:
  Go version: go1.23.7 (linux, amd64)
```

```suggestion
    CGO_ENABLED=0 GOOS=linux go build -trimpath -ldflags '-s -w' -o /icingadb ./cmd/icingadb
```

```
docker run -it --rm icingadb /icingadb --version
Icinga DB version: 1.2.1-g1ec7f60-dirty

Build information:
  Go version: go1.23.7 (linux, amd64)
  Git commit: 1ec7f606d526e1157b5d7851ceb8fddae42b2b14 (modified)
```

This was a tricky one. Thanks to https://github.com/golang/go/issues/51264#issuecomment-1048335147 for pointing me into this direction.

---

Wow, thanks for this find and the fix. To be honest, I wouldn't have checked the version.